client/modules/User/pages/AccountView.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { PropTypes } from 'react';
+import { reduxForm } from 'redux-form';
+import { bindActionCreators } from 'redux';
+import { browserHistory } from 'react-router';
+import InlineSVG from 'react-inlinesvg';
+import axios from 'axios';
+import { updateSettings } from '../actions';
+import AccountForm from '../components/AccountForm';
+
+const exitUrl = require('../../../images/exit.svg');
+const logoUrl = require('../../../images/p5js-logo.svg');
+
+
+class AccountView extends React.Component {
+  constructor(props) {
+    super(props);
+    this.closeAccountPage = this.closeAccountPage.bind(this);
+    this.gotoHomePage = this.gotoHomePage.bind(this);
+  }
+
+  closeAccountPage() {
+    browserHistory.push(this.props.previousPath);
+  }
+
+  gotoHomePage() {
+    browserHistory.push('/');
+  }
+
+  render() {
+    return (
+      <div className="form-container">
+        <div className="form-container__header">
+          <button className="form-container__logo-button" onClick={this.gotoHomePage}>
+            <InlineSVG src={logoUrl} alt="p5js Logo" />
+          </button>
+          <button className="form-container__exit-button" onClick={this.closeAccountPage}>
+            <InlineSVG src={exitUrl} alt="Close Account Page" />
+          </button>
+        </div>
+        <div className="form-container__content">
+          <h2 className="form-container__title">My Account</h2>
+          <AccountForm {...this.props} />
+          {/* <h2 className="form-container__divider">Or</h2>
+          <GithubButton buttonText="Login with Github" /> */}
+        </div>
+      </div>
+    );
+  }
+}
+
+function mapStateToProps(state) {
+  return {
+    initialValues: state.user, // <- initialValues for reduxForm
+    user: state.user,
+    previousPath: state.ide.previousPath
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return bindActionCreators({ updateSettings }, dispatch);
+}
+
+function asyncValidate(formProps, dispatch, props) {
+  const fieldToValidate = props.form._active;
+  if (fieldToValidate) {
+    const queryParams = {};
+    queryParams[fieldToValidate] = formProps[fieldToValidate];
+    queryParams.check_type = fieldToValidate;
+    return axios.get('/api/signup/duplicate_check', { params: queryParams })
+      .then((response) => {
+        if (response.data.exists) {
+          const error = {};
+          error[fieldToValidate] = response.data.message;
+          throw error;
+        }
+      });
+  }
+  return Promise.resolve(true).then(() => {});
+}
+
+function validate(formProps) {
```

Comment:
since login/signup use the same regexes to do validation, it probably makes sense to set them to some other variable in another file and import them. I've played around with the idea of converting the `constants.js` file to a folder called `constants`, then changing `constants.js` to `actiontypes.js`, and then, more constant files can be added as necessary. Or maybe there's a better place to put this file.